crates/ironrdp-cliprdr-format/src/bitmap.rs

```diff
@@ -237,22 +232,27 @@ impl BitmapInfoHeader {
             clr_important,
         };
 
-        header.validate_invariants()?;
-
         Ok((header, size))
     }
 
+    // INVARIANT: output (width) < 10_000
     fn width(&self) -> u16 {
-        // Cast is safe, invariant is checked in `encode_with_size` and `decode_with_size`.
-        u16::try_from(self.width.abs()).unwrap()
+        let abs = self.width.abs();
+        debug_assert!(abs < 10_000);
+        // Per the invariant on self.width, this cast is infaillible.
+        u16::try_from(abs).unwrap()
     }
 
+    // INVARIANT: output (height) < 10_000
     fn height(&self) -> u16 {
-        // Cast is safe, invariant is checked in `encode_with_size` and `decode_with_size`.
-        u16::try_from(self.height.abs()).unwrap()
+        let abs = self.height.abs();
+        debug_assert!(abs < 10_000);
+        // Per the invariant on self.height, this cast is infaillible.
+        u16::try_from(abs).unwrap()
     }
 
-    fn flip_vertically(&self) -> bool {
```

Comment:
Oof, I really hate how I named this method, IDK how it slipped through my self-review, thanks for fixing that!  👍